app/screens/auth/SwitchNetwork.tsx

```diff
@@ -105,18 +109,27 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
 
   const goNext = (genesisID: string | undefined) => {
     if (creatingWallet) {
-      if (isWalletOnly && genesisID?.length)
+      if (isWalletOnly && genesisID?.length) {
         return history.push(AuthPath.ConnectToAPI, {
-          redirect: AuthPath.CreateWallet,
+          redirect: AuthPath.ProtectWalletMnemonicStrength,
           genesisID,
           isWalletOnly,
           creatingWallet,
           mnemonic,
         });
-      return history.push(AuthPath.CreateWallet, {
+      }
+
+      if (mnemonic?.existing) {
+        return history.push(AuthPath.CreateWallet, {
+          genesisID,
+          isWalletOnly,
+          mnemonic,
+        });
+      }
+
```

Comment:
[Still no comment there](https://github.com/spacemeshos/smapp/pull/1429#discussion_r1287120035) :-\

Hopefully, it is a part of restoring the wallet...